### DISCUSSION TITLE
Is this possible: Automatically turn relay Off when MQTT is down long time?

### DISCUSSION THREAD

**Discussion by miXza81** [arendst/Tasmota#14946] [Q&A] [answered]
Hi.

Any tips how I should configure Tasmota when I want it turn relay Off if there is over 10 minutes lasting error when trying to communicate with MQTT broker? Examples: Maybe MQTT broker is down, or there is no Wifi connection…

**Comment by barbudor** [comment]
You need to read https://tasmota.github.io/docs/Rules/
This is also similar request here in Discussions

**Comment by miXza81** [comment, marked as answer]
I have wrote this code...

```
Rule1 ON Mqtt#Disconnected DO Backlog ruletimer1 600 ENDON ON Mqtt#Connected DO ruletimer1 0 ENDON ON rules#timer=1 DO Power1 0 ENDON
rule1 1
```

Can I trust that it will work fine in every situation or have I missed something? (It's safety feature)

**Reply by barbudor** [reply]
Good job, I would have done the same 👍
The backlog in the 1st rule is not required

**Reply by sfromis** [reply]
While that rule set should work when it loses MQTT connection, if Tasmota boots without getting MQTT connection, and with the relay state on, nothing will be done.
